tests/test_pixeldataset.py

```diff
@@ -156,6 +172,94 @@ def test_pixeldataset_from_file_parquet(setup_basic_pixel_dataset, tmp_path):
     )
 
 
+def test_pixeldataset_from_file_parquet_backward_comp_with_pyarrow_types(
+    setup_basic_pixel_dataset, tmp_path
+):
+    (
+        dataset,
+        edgelist,
+        adata,
+        metadata,
+        polarization_scores,
+        colocalization_scores,
+    ) = setup_basic_pixel_dataset
+
+    # When reading the old file-format we will fall back to the
+    # polars file reader
+    with mock.patch(
+        "pixelator.pixeldataset.pd.read_parquet",
+        side_effect=ValueError(),
+    ):
+        file_target = tmp_path / "dataset.pxl"
+        dataset.save(str(file_target))
+        dataset_new = PixelDataset.from_file(str(file_target))
+
+        assert_frame_equal(edgelist, dataset_new.edgelist, check_categorical=False)
+        assert_frame_equal(
+            edgelist,
+            dataset_new.edgelist_lazy.collect().to_pandas(),
+            check_categorical=False,
+            check_dtype=False,
+        )
+
+        assert_frame_equal(
+            adata.to_df(),
+            dataset_new.adata.to_df(),
+        )
+
+        assert metadata == dataset_new.metadata
+
+        assert_frame_equal(
+            polarization_scores, dataset_new.polarization, check_dtype=False
```

Comment:
So, the reason for this is that we were using the `pyarrow` extension when reading these data frames (and then not converting them). I dropped this now, since especially for these smaller data frames it doesn't make that much of a difference in read performance/memory usage. We might go back to using `pyarrow` more extensively in the future, but I think we should wait for the rest of the ecosystem to catch up first.